web/src/components/hooks/useDownloadValues.test.jsx

```diff
@@ -0,0 +1,163 @@
+      };
+
+      expect(result.current.variables).toEqual(undefined);
+      expect(fetchValuesSpy).toHaveBeenCalledTimes(1);
+      expect(fetchValuesSpy).toHaveBeenCalledWith(expectedFetchConfig);
+    });
+    it.todo(
+      "test the rest of this or figure out how to use react-query with it"
+    );
+  });
+  describe("getValues", () => {
+    it("calls fetch with the correct url and configuration", async () => {
+      const expectedBody = {
+        test: "test",
+      };
+      const blobSpy = jest.fn(() => Promise.resolve(expectedBody));
+      const _fetchValuesSpy = jest.fn(() =>
+        Promise.resolve({
+          ok: true,
+          blob: blobSpy,
+        })
+      );
+      const testAppSlug = "testAppSlug";
+      const testToken = "testToken";
+      const testAPIEndpoint = "testAPIEndpoint";
+      const testGetValuesConfig = {
+        _fetch: _fetchValuesSpy,
+        _token: testToken,
+        apiEndpoint: testAPIEndpoint,
+        appSlug: testAppSlug,
+      };
+
+      const expectedAPIEndpoint = `${testAPIEndpoint}/app/${testAppSlug}/values`;
+      const expectedResponse = {
+        data: expectedBody,
+      };
+      const expectedFetchConfig = {
+        method: "GET",
+        headers: {
+          Authorization: testToken,
+          "Content-Type": "application/blob",
+        },
+      };
+      await expect(getValues(testGetValuesConfig)).resolves.toEqual(
+        expectedResponse
+      );
+      expect(_fetchValuesSpy).toHaveBeenCalledTimes(1);
+      expect(_fetchValuesSpy).toHaveBeenCalledWith(
+        expectedAPIEndpoint,
+        expectedFetchConfig
+      );
+      expect(blobSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it("throws error when response is not ok", async () => {
+      const _fetchValuesSpy = jest.fn(() =>
+        Promise.resolve({
+          ok: false,
+        })
+      );
+      const testAppSlug = "testAppSlug";
+
+      const testToken = "testToken";
+      const testAPIEndpoint = "testAPIEndpoint";
+      const testGetValuesConfig = {
+        _fetch: _fetchValuesSpy,
+        _token: testToken,
+        apiEndpoint: testAPIEndpoint,
+        appSlug: testAppSlug,
+      };
+
+      await expect(getValues(testGetValuesConfig)).rejects.toThrowError(
+        "Error fetching values"
+      );
+    });
+
+    it("throws error when response is not blob", async () => {
+      const _fetchValuesSpy = jest.fn(() =>
+        Promise.resolve({
+          ok: true,
+          blob: () => Promise.reject(new Error("Error parsing blob")),
+        })
+      );
+
+      const testAppSlug = "testAppSlug";
+      const testToken = "testToken";
+      const testAPIEndpoint = "testAPIEndpoint";
+      const testGetValuesConfig = {
+        _fetch: _fetchValuesSpy,
+        _token: testToken,
+        apiEndpoint: testAPIEndpoint,
+        appSlug: testAppSlug,
+      };
+
+      await expect(getValues(testGetValuesConfig)).rejects.toThrowError(
+        "Error parsing blob"
```

Comment:
copilot taught me this approach to testing async functions where you use `await expect(call).reject.toThrowError()`